Tests/Runtime/Shape/BezierMeshTests.cs

```diff
@@ -0,0 +1,188 @@
+            m_Mesh.faceCountPerSegment + m_Mesh.faceCountPerSegment;
+
+        Assert.That(m_Mesh.mesh.vertexCount, Is.EqualTo(newExpectedVertexCount));
+    }
+
+    [Test]
+    public void BezierMesh_ChangingFaceCount_ExtrudesMeshCorrectly()
+    {
+        Assert.That(m_Mesh, Is.Not.Null);
+
+        var totalSegmentCount = (int)m_Mesh.splineContainer.Splines[0].GetLength() * m_Mesh.segmentsPerUnit;
+        var faceCount = m_Mesh.faceCountPerSegment * totalSegmentCount;
+        var expectedFaceCount = m_Mesh.faceCountPerSegment * totalSegmentCount;
+
+        Assert.That(faceCount, Is.EqualTo(expectedFaceCount));
+
+        m_Mesh.faceCountPerSegment = 10;
+        m_Mesh.ExtrudeMesh();
+        var newTotalSegmentCount = (int)m_Mesh.splineContainer.Splines[0].GetLength() * m_Mesh.segmentsPerUnit;
+        var newFaceCount = m_Mesh.faceCountPerSegment * totalSegmentCount;
+        var newExpectedFaceCount = m_Mesh.faceCountPerSegment * newTotalSegmentCount;
+
+        Assert.That(newFaceCount, Is.EqualTo(newExpectedFaceCount));
+    }
+
+    [Test]
+    public void BezierMesh_SplineWithLessThanTwoKnots_DoesNotExtrudeMesh()
+    {
+        Assert.That(m_Mesh, Is.Not.Null);
+
+        m_Mesh.splineContainer.Splines[0].Knots = new List<BezierKnot> { new BezierKnot(float3.zero) };
+        m_Mesh.ExtrudeMesh();
+
+        Assert.That(m_Mesh.mesh.vertexCount, Is.EqualTo(0));
+    }
+
+    [Test]
+    public void BezierMesh_HavingMultipleSplinesInTheContainer_ExtrudesMeshOnAllSplines()
+    {
+        Assert.That(m_Mesh, Is.Not.Null);
+
+        var splines = new List<Spline>();
+
+        var p1 = float3.zero;
+        var p2 = new float3(5f, 0f, 0f);
+        splines.Add(new Spline(new List<BezierKnot>()
+        {
+            new BezierKnot(p1),
+            new BezierKnot(p2)
+        }));
+
+        var p3 = new float3(10f, 0f, 0f);
+        var p4 = new float3(15f, 0f, 0f);
+        splines.Add(new Spline(new List<BezierKnot>()
+        {
+            new BezierKnot(p3),
+            new BezierKnot(p4)
+        }));
+
+        m_Mesh.splineContainer.Splines = splines;
+        Assert.That(m_Mesh.splineContainer.Splines.Count(), Is.EqualTo(2));
+
+        m_Mesh.ExtrudeMesh();
+
+        var vertices = m_Mesh.mesh.GetVertices();
+        var isVertexFoundAtP1 = false;
+        var isVertexFoundAtP2 = false;
+        var isVertexFoundAtP3 = false;
+        var isVertexFoundAtP4 = false;
+
+        foreach (var vertex in vertices)
+        {
+            var pos = vertex.position;
+
+            if (Mathf.Approximately(pos.x, p1.x))
+                isVertexFoundAtP1 = true;
+            if (Mathf.Approximately(pos.x, p2.x))
+                isVertexFoundAtP2 = true;
+            if (Mathf.Approximately(pos.x, p3.x))
+                isVertexFoundAtP3 = true;
+            if (Mathf.Approximately(pos.x, p4.x))
+                isVertexFoundAtP4 = true;
+        }
```

Comment:
As you know exactly how many vertices there are and how they are ordered, I would prefer to directly check the vertex count and some vertex positions. For instance:
`Assert.That((Mathf.Approximately(vertices[0].x, p1.x), Is.True);`